### PR TITLE
Use JSON to return an ArrayList to a fetch function.

### PR DESCRIPTION
**Pull request by SydneyCaulfeild**
I have added an ArrayList to the DataServlets.java file and used JSON to return the ArrayList from the server. I also added a link to the main page in the navigation bar.

### CODE REVIEW

**Review by Vector-S — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,9 +26,35 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> canadianCities;
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: It is generally better to declare variables as interfaces, so "List<String>". Declaring as specific implementation can get you into trouble later.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,9 +26,35 @@
...
+
+  @Override
+  public void init(){
+    canadianCities = new ArrayList<>();
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: It looks it is better to initialize this above where it is declared.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -24,9 +26,35 @@
...
+    canadianCities.add("Canmore, Alberta"); 
+    canadianCities.add("St. Johns, Newfoundland"); 
+    canadianCities.add("Kingston, Ontario"); 
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello and welcome to my portfolio!");
+    //response.setContentType("text/html;");
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: please don't leave commented out code. You can always find it in history if needed again.

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -7,7 +7,7 @@
   <link rel="stylesheet" href="style.css">
   <script src="script.js"></script>
 </head>
-<body onload="fetchWelcome()"> 
+<body onload="fetchList()">
```
Comment (lipczak-g) [inline portfolio/src/main/webapp/index.html]: fetchFavoriteCities() sounds like a better name. Overall, the type is less important than the content.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -17,14 +17,25 @@ <h1>Sydney Caulfeild</h1>
   <!-- Element where the heading's navigation bar will be inserted. -->
   <header></header>
  
-  <div id="welcome-container"></div>
+  <!-- <div id="welcome-container"></div> -->
```
Comment (lipczak-g) [inline portfolio/src/main/webapp/index.html]: no commented out code :)

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -47,8 +47,31 @@ fetch("footer.html")
...
   });
+}*/
+
+//fetching the JSON arraylist string from server
+function fetchList() {
+  fetch('/data').then(response => response.json()).then((citiesList) => {
+    console.log(citiesList);
+    const citiesListElement = document.getElementById('cities-list-container');
+    citiesListElement.innerHTML = '';
+    citiesListElement.appendChild(
```
Comment (lipczak-g) [inline portfolio/src/main/webapp/script.js:62]: this should be done in a loop, there is no guarantee you will get 3 items back.

Comment (SydneyCaulfeild) [reply portfolio/src/main/webapp/script.js:62]: Ok, I have implemented a For Each loop instead!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (SydneyCaulfeild) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Sounds good! I've declared it as a type List instead.

**Pull request merged**